cmake/OpenDDSConfig.cmake

```diff
@@ -302,53 +306,71 @@ macro(opendds_vs_force_static)
   endforeach()
 endmacro()
 
-set(_suffix_RELEASE "")
-set(_suffix_DEBUG d)
-
 if(MSVC AND OPENDDS_STATIC)
   opendds_vs_force_static()
-
-  set(_suffix_RELEASE s${_suffix_RELEASE})
-  set(_suffix_DEBUG s${_suffix_DEBUG})
 endif()
 
-foreach(_cfg  RELEASE  DEBUG)
-  set(_sfx ${_suffix_${_cfg}})
-
-  foreach(_lib ${_ace_libs})
-    string(TOUPPER ${_lib} _LIB_VAR)
-
-    find_library(${_LIB_VAR}_LIBRARY_${_cfg}
-      ${_lib}${_sfx}
-      HINTS ${_ace_lib_hints}
-    )
-  endforeach()
-
-  foreach(_lib ${_tao_libs})
-    string(TOUPPER ${_lib} _LIB_VAR)
-
-    find_library(${_LIB_VAR}_LIBRARY_${_cfg}
-      ${_lib}${_sfx}
-      HINTS ${_tao_lib_hints}
-    )
-  endforeach()
-
-  foreach(_lib ${_opendds_libs})
-    string(TOUPPER ${_lib} _LIB_VAR)
-
-    find_library(${_LIB_VAR}_LIBRARY_${_cfg}
-      ${_lib}${_sfx}
-      HINTS ${_dds_lib_hints}
-    )
-  endforeach()
+function(opendds_find_our_libraries_for_config config suffix)
+  if(MSVC AND OPENDDS_STATIC)
+    set(suffix "s${suffix}")
+  endif()
 
-endforeach()
+  macro(find_library_group lib_group_name libs)
+    set(lib_dir "${${lib_group_name}_LIB_DIR}")
+
+    foreach(lib ${libs})
+      set(lib_file_base "${lib}${suffix}")
+      string(TOUPPER ${lib} var_prefix)
+      set(lib_var "${var_prefix}_LIBRARY_${config}")
+
+      find_library(${lib_var} "${lib_file_base}" HINTS "${lib_dir}")
+      set(found_var "${var_prefix}_LIBRARY_FOUND")
+      if(${lib_var})
+        set(${found_var} TRUE PARENT_SCOPE)
+      elseif(NOT DEFINED ${found_var})
+        set(${found_var} FALSE PARENT_SCOPE)
+      endif()
+
+      if(MSVC AND NOT OPENDDS_STATIC)
+        # find_library finds the ".lib" file on Windows, but if OpenDDS is not
+        # static we also need the ".dll" file for IMPORTED_LOCATION and
+        # IMPORTED_RUNTIME_ARTIFACTS to work correctly.
+        find_file("${lib_var}_DLL" "${lib_file_base}.dll" HINTS "${lib_dir}")
+      endif()
+    endforeach()
+  endmacro()
+
+  find_library_group("ACE" "${_ace_libs}")
+  find_library_group("TAO" "${_tao_libs}")
+  find_library_group("OPENDDS" "${_opendds_libs}")
+endfunction()
+
+if(MSVC)
+  opendds_find_our_libraries_for_config("RELEASE" "")
+  opendds_find_our_libraries_for_config("DEBUG" "d")
+elseif(OPENDDS_DEBUG)
+  opendds_find_our_libraries_for_config("DEBUG" "")
+else()
+  opendds_find_our_libraries_for_config("RELEASE" "")
+endif()
 
 include(SelectLibraryConfigurations)
 
 foreach(_lib ${_all_libs})
   string(TOUPPER ${_lib} _LIB_VAR)
   select_library_configurations(${_LIB_VAR})
+
+  # These paths might be symlinks and IMPORTED_RUNTIME_ARTIFACTS seems to not
+  # work correctly with symlinks, so resolve them now.
```

Comment:
I didn't think about that, but maybe I should looks for an issue or make if there isn't one. The issue was it copied the symlinks instead of the actual files.

Reply:
I have created the issue: https://gitlab.kitware.com/cmake/cmake/-/issues/23249 and will note it in the comment.